src/foam/lib/json/JSONParser.java

```diff
@@ -25,9 +25,10 @@ public FObject parseString(String data, Class defaultClass) {
     ps.setString(data);
     ParserContext x = new ParserContextImpl();
     x.set("X", getX());
-    x.set("defaultClass", defaultClass);
+    if ( defaultClass != null ) {
+      parser = new ExprParser(defaultClass);
```

Comment:
This makes this class be non reusable since calling parseString() has side-effects (ie. rewriting 'parser').
A safer approach would be to change the line below to:
 ps = (StringPStream) ps.apply(defaultClass == null ? parser : new ExprParser(defaultClass), x);

Reply:
Thanks, done.